### PR TITLE
Patched HIGH CVE-2022-38725 for syslog-ng

### PR DESCRIPTION
**Pull request by mbykhovtsev-ms**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Patched high CVE-2022-38725 for `syslog-ng`. Patch found at https://github.com/syslog-ng/syslog-ng/pull/4110

###### Change Log  

- Patched high CVE-2022-38725 for `syslog-ng`

###### Does this affect the toolchain?  

**NO**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2022-38725

###### Test Methodology

- Fast Track PR check

### CODE REVIEW

File: SPECS/syslog-ng/syslog-ng.spec [hunk 1]
```diff
@@ -50,7 +51,7 @@ Requires:       %{name} = %{version}-%{release}
  needed to build applications using syslog-ng APIs.
 
 %prep
-%setup -q
+%autosetup
```
Comment (PawelWMS) [inline SPECS/syslog-ng/syslog-ng.spec:54]: You still need to add `-p1` or `-pX` in general to apply patches.

**Comment by PawelWMS** [comment]
@mbykhovtsev-ms, just a heads-up: all PRs against `fasttrack/2.0` should be rebased, since we finished a monthly update, which also updated `fasttrack/2.0`. Rebasing (or merging) the latest state of the branch will re-trigger the PR check and allow us to validate that the latest state of the repo still builds correctly with the changes from the PRs.

**Comment by Redent0r** [comment]
Wondering if I can do https://github.com/microsoft/CBL-Mariner/pull/6431/commits/bbeb3226bdb69b46e9fdaf6b079de8e35ae20607?

Here's what happened:
- When `make check` was included, fast track build failed due to a ptest return code. The error message is "Unit tests are disabled"
- Looks like it comes from https://github.com/syslog-ng/syslog-ng/blob/syslog-ng-3.33.2/Makefile.am#L138 and they can be enabled
- I wasn't able to enable the flag despite trying https://github.com/microsoft/CBL-Mariner/commit/92ee41dca9571800d7fcb7b3687b32073039782a and https://github.com/microsoft/CBL-Mariner/commit/01fd2523cf59a6758864f2216bd88d429fa3d3e2
- So I try to check how Fedora does `make check`. I try to look at https://src.fedoraproject.org/rpms/syslog-ng/blob/epel9/f/syslog-ng.spec and I see there's no check section and it actually got removed (https://src.fedoraproject.org/rpms/syslog-ng/blob/epel9/f/syslog-ng.spec#_653)

**Comment by PawelWMS** [comment]
> Wondering if I can do [bbeb322](https://github.com/microsoft/CBL-Mariner/commit/bbeb3226bdb69b46e9fdaf6b079de8e35ae20607)?
> 
> Here's what happened:
> 
> * When `make check` is included, fast track build failed due to a ptest return code. The error message is "Unit tests are disabled". We can also see this error when buddy building over fasttrack/2.0 https://dev.azure.com/mariner-org/mariner/_build/results?buildId=456539&view=ms.vss-test-web.build-test-results-tab&runId=145823&resultId=100000&paneView=debug
> * Looks like it comes from https://github.com/syslog-ng/syslog-ng/blob/syslog-ng-3.33.2/Makefile.am#L138 and they can be enabled
> * I wasn't able to enable the flag despite trying [92ee41d](https://github.com/microsoft/CBL-Mariner/commit/92ee41dca9571800d7fcb7b3687b32073039782a) and [01fd252](https://github.com/microsoft/CBL-Mariner/commit/01fd2523cf59a6758864f2216bd88d429fa3d3e2)
> * So I try to check how Fedora does `make check`. I try to look at https://src.fedoraproject.org/rpms/syslog-ng/blob/epel9/f/syslog-ng.spec (3.35.1 theirs vs 3.33.2 ours) and I see there's no check section and it actually got removed (https://src.fedoraproject.org/rpms/syslog-ng/blob/epel9/f/syslog-ng.spec#_653)
> * Additionally I noticed `make check` calls [check target](https://github.com/syslog-ng/syslog-ng/blob/syslog-ng-3.33.2/Makefile.am#L134), which in turn calls [check_target_guard](https://github.com/syslog-ng/syslog-ng/blob/syslog-ng-3.33.2/Makefile.am#L136). check_target_guard seems to only check if unit test flag is passed and print error if not. Doesn't start or run any tests it seems
> 
> Passing fasttrack build without `make check`: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=456990&view=results

I took a look at the `configure` file in the sources and found this:
```
 if test "$enable_tests" != "no"; then
  ENABLE_TESTING_TRUE=
  ENABLE_TESTING_FALSE='#'
else
  ENABLE_TESTING_TRUE='#'
  ENABLE_TESTING_FALSE=
fi
```

It seems, that `enable_tests` must be set to something else than `no` but on a different line:
```
```

**Review by 0xba1a — APPROVED** [review]
As Pawel mentioned, please create P1 bug and provide the link in this PR for reference

**Comment by Redent0r** [comment]
Created [P1 bug](https://microsoft.visualstudio.com/OS/_workitems/edit/47882516) and linked back to this PR. Also added back the `make check` to the %check section and commented it out with a todo comment also linking to this PR. Thanks!

**Pull request merged**